homeassistant/util/temperature.py

```diff
@@ -9,3 +10,17 @@ def fahrenheit_to_celsius(fahrenheit: float) -> float:
 def celsius_to_fahrenheit(celsius: float) -> float:
     """Convert a Celsius temperature to Fahrenheit."""
     return celsius * 1.8 + 32.0
+
+
+def convert(temperature: float, from_unit: str, to_unit: str) -> (float, str):
+    """Convert a temperature from one unit to another."""
+    if from_unit not in [TEMP_CELSIUS, TEMP_FAHRENHEIT] or \
+        to_unit not in [TEMP_CELSIUS, TEMP_FAHRENHEIT] or \
+            to_unit == from_unit:
+        # If no need or unknown conversion, return temperature
+        return temperature, from_unit
```

Comment:
Why are we now returning the unit of measurement when doing a conversion? As long as we have validation on our config inputs, we should never ever get an unknown unit. In fact, if we get an unknown unit I think we should blow up, raise a `ValueError` instead of silently accepting it.